server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -71,10 +73,14 @@ private void handleUpdate(String mailboxId, MailboxUpdateRequest updateRequest,
         try {
             validateMailboxName(updateRequest, mailboxSession);
             Mailbox mailbox = getMailbox(mailboxId, mailboxSession);
+            checkRole(mailbox.getRole());
```

Comment:
For me preconditions should be the first-of-all process done by one method.

In our case, we can do this for some checks (mailbox name), 
but then we came at a point where we need some data from the current persisted state.

The we need a process before the remaining preconditions, and that's why it's not precondition and Exceptions are here for, IMO.